bitforge/network.py

```diff
@@ -0,0 +1,127 @@
+from collections import namedtuple
+from errors import BitforgeError, KeyValueError, StringError
+
+
+# TODO: document these fields
+# TODO: which should be unique? It should be enforced
+BaseNetwork = namedtuple('Network', [
+    # Descriptors
+    'name',
+    'aliases',
+    # Serialization magic numbers
+    'pubkeyhash',
+    'wif_prefix',
+    'scripthash',
+    'hd_public_key',
+    'hd_private_key',
+    'magic',
+    # Network parameters
+    'port',
+    'seeds',
+])
+
+
+class Network(BaseNetwork):
+    """Parameters of a Bitcoin-compatible network."""
+
+    class Error(BitforgeError):
+        pass
+
+    class UnknownNetwork(Error, KeyValueError):
+        """No network with '{key}' matching '{value}'"""
+
+    class MultipleNetworks(Error, KeyValueError):
+        """Multiple networks with '{key}' matching '{value}'"""
+
+    class ExistingNetworkName(Error, StringError):
+        """Network name '{string}' already taken"""
+
+    _networks = {}
+
+    @classmethod
+    def get(cls, name, raises=True):
+        """Get a network by name or alias."""
+
+        network = cls._networks.get(name, None)
+
+        if raises and network is None:
+            raise Network.UnknownNetwork('name', name)
+
+        return network
+
+    # TODO: discuss if this method should exist at all
```

Comment:
The main use case for this is parsing some serialization formats. This makes registering new networks/altcoins quite easily without changing the parsing code.

It could be a method or an static function in the network module. Not sure what's better.